School professions [CR]

AllisonBW opened the pull request:
I added several student professions, intended for/inspired by Jakers' School of the Dead scenario idea, but available for other scenarios in this version of the PR. I added the Elementary Student (elementary_student), Skater Boy/Girl (skaterkid), Student of Faith (churchkid), Juvenile Delinquent (jdelinquent), Preppy Student (prep_student), Survivalist Jr. (jr_survivalist), Bionic Student (bionic_student), Dodgeball Player (dodgeball_player), Science Club Member (science_club_mem), and A/V Club Member (av_club_mem) to the professions json, and also added them to some of the whitelists in scenarios.json: all of them were added to the whitelist for Burning Building; all of them except for Survivalist Jr. were added to the whitelist for Infected; and Survivalist Jr. was added to the whitelist for Wilderness.

I made the edits in Notepad++ and tested and debugged them locally until they were showing up in the right circumstances and chargen wasn't throwing errors at me. I may have then committed a mortal sin by uploading my changes to the files in my branch by way of the GitHub web interface.

The balance/flavouring/etc. could probably use some critique, especially the Dodgeball Player and club members, which are the newest and haven't had dev commentary/criticism yet. I also have a bit of a hangup with the Juvenile Delinquent and the hairpins; I included them with the intent that the JD has them on their person for lockpicking purposes, but they spawn _wearing_ the hairpins, which is probably a minor nitpick. Also, I kind of want to add a student that's -1 point; it'd probably be like the Elementary Student minus the backpack, lunch, and book, and maybe the scarf and/or jacket. I just don't know what I'd call it.

Sorry if I screwed something up in here somewhere. First time coding, first PR, first time using Git, and all that.

narc0tiq commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,368 @@
+                    "tshirt",
+                    "socks",
+                    "backpack",
+					"sneakers",
```
Looks like some Tab characters have snuck into your changes here -- tell your Notepad++ that .json files should have spaces instead of tabs (four spaces is probably correct).

narc0tiq commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,368 @@
+        "type": "profession",
+        "ident": "av_club_mem",
+        "name": "A/V Club Member",
+        "description": "You were a member of the school A/V club. Are those even a thing in 2042? You're sure there's some way you can use your technical skills to help stay alive. You just haven't figured out how to make an awesome death ray yet.",
```
Maybe omit the rhetorical question ('Are those even a thing in 2042?'). Clearly, they are, else this profession wouldn't exist.

AllisonBW commented:
You ain't kiddin'. Lots of tabs in there. I'll have to go back through and fix it up. Also, point noted and taken about the rhetorical question.

narc0tiq commented:
Balance-wise, it's very odd that one point of speech ("Preppy Student") is more valuable than a point of electronics and one of computer skills ("A/V Club Member"). Nothing else jumped out at me while skimming.

AllisonBW commented:
Actually, the Preppy Student's point isn't because of the point of Speech. It's actually about the messenger bag (pretty good starting storage item, what with the 0 enc) and the Adderall, which gives that particular profession a huge advantage in escaping early-game conflicts alive.

Thanks for checking out my PR and commenting on it, by the by!

narc0tiq commented:
> It's actually about the messenger bag [...] and the Adderall [...]

Oh! Okay, that makes perfect sense to me. Good stuff, thank you!

AllisonBW commented:
OK, got it cleaned up. How's it look now?

narc0tiq commented:
Shiny!

AllisonBW commented:
Thank you kindly, narc0tiq! Sorry about the whole using the web interface thing.

narc0tiq commented:
It's not something you need to apologize for, it's just that it has caused
issues before with other people's PRs, and I wouldn't want to see yours not
being considered because of that. This one looks to be behaving fine.

On Mon, Jan 26, 2015 at 4:37 PM, AllisonBW <notifications@github.com> wrote:

> Thank you kindly, narc0tiq! Sorry about the whole using the web interface
> thing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/11000#issuecomment-71469522>
> .
>

Snaaty commented:
These look nice! 
Next thing we need is probably a refurbished school and a corresponding school-scenario, though!

AllisonBW commented:
Thank you, Snaaty. And that's actually the intention! http://smf.cataclysmdda.com/index.php?topic=9286.0

That said, any thoughts on the thing with the JD spawning wearing the pair of hairpins, or what to call the student with the totally unprepared start? I know there's no fix from the former aside from "just leave it" or "remove the hairpins or replace them with something else" (maybe a couple of improvised picklocks instead, but those are heavier and I don't know if they're more or less effective than hairpins).

KA101 commented:
Yeah, most school kids should be kept in school, but we can add that when the school scenario lands.  I didn't see anything terribly problematic yesterday, but then that was yesterday.

KA101 commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,368 @@
+                "jeans",
+                "socks",
+                "lowtops",
+                "runner_bag",
```
Runner bag is a pretty decent starting storage (2 less volume than the mbag).  IMO it doesn't *quite* warrant a second point but that's open to discussion.

KA101 commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,368 @@
+                "gloves_fingerless",
+                "matches",
+                "hairpin",
+                "hairpin",
```
Obviously you're wearing the hairpins so mean Mr. KA101 doesn't take them away for being lockpicking tools!  ;-)

KA101 commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,368 @@
+                "name": "electronics"
+            },
+            {
+                "level": 1,
```
That's a chargen point worth of skills there.  Cost +1.

AllisonBW commented:
Thank you for the critique, KA101! I downgraded the Dodgeball Player's runner pack to a fanny pack, which should put it a little more safely in the 1-point zone, and changed the cost of the A/V Club Member to 1.

A few things:

1. As for the JD and hairpins--good rationale. However, I'm still picturing Little Mr. Tough Guy wearing a couple of hairpins in his hair and while adorable and kind of a funny image I dunno that it fits. Would there be any objection if I switched the hairpins out for a couple of improvised picklocks? I don't know what difference there is, if any, in the lockpicking ability of hairpins and improvised picklocks; they both strike me as decidedly, uh, "improvised."

2. The Student of Faith just doesn't seem to have much of an identity or gameplay niche at the moment. Would there be any objection if I dropped it?

3. I'm considering the names "Unprepared Student" (kind of unwieldy) or "Poor Student" for the variant elementary student that loses the backpack, lunch, book, and probably the scarf to get a -1 point cost. Any thoughts on this?

In the meantime, I'm going to see about uploading Jakers's janitor, teacher, and gym teacher next.

AllisonBW commented:
Added Jakers's professions, both to professions.json and to scenarios.json. Janitor, Teacher, and Gym Teacher, all of which are now in the general pool and the lists for the Infected and Burning Building scenarios (which should probably be reversed once the school scenario's in). I *think* I priced the Janitor and Teacher correctly, but I'm not 100%; I kept the Gym Teacher's price as Jakers set it, and I'm not certain of that either. I'll ask Jakers how he feels about dropping the janitor's t-shirt, light jacket, and pants for a jumpsuit; that might serve to give it a little difference from the unprepared-student start if that gets in.

KA101 commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,460 @@
+        "ident": "janitor",
+        "name": "Janitor",
+        "description": "You earned a living from sweeping up chocolate wrappers and picking chewing gum from under tables. Now the only thing you'll be sweeping are the brains of the dead.",
+        "points": -1,
```
No significant* carrying capacity, lightweight clothing, and no knife/fire.  This one might qualify for -2.

*As in, I'm thinking you'd have to wield the mop in order to carry anything else effectively.  Same problem the various athletic folks have.

KA101 commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,460 @@
+        "skills": [
+            {
+                "level": 3,
+                "name": "speech"
```
Speech may not be all that useful, but this would cost 2 chargen points to buy (first chargen point buys rank 2, next buys rank 3).  You're going to need to pay at least 1 for the profession.

KA101 commented on data/json/professions.json:
```diff
@@ -3417,5 +3417,460 @@
+        "items": {
+            "both": [
+                "baseball",
+                "bat",
```
Wooden baseball bat is an excellent starting weapon.  Weapon + carry capacity from the runner bag are likely worth a point, in addition to the point each in Dodge and Speech.

KA101 commented:
1) They're equivalent, but the hairpin is much easier to carry.  Net minor nerf and I'd only issue one set.  That said, girls/women can be delinquent too!

2) None on my end.  Folks want to go religious, fine, but we don't start them as such.

3) Losing the chem set is a good idea; possibly take a point in Mechanics.  Chem textbook isn't a bad idea.  How about a Vo-Tech student who's got a point each in Fabrication and Mechanics?  (Vocational-Technical school, a USIan thing where they start kids on the trades.  I flunked the entry exam as my fine-motor skills weren't to spec.)

4) Meh, Unprepared sounds as though xe's missing xyr books or something.

Jumpsuit isn't typical janitor dress IME but agreed that it'd change things up.

AllisonBW commented:
Thank you again, KA101!

I think giving the Janitor a point value of -2 might be a bit extreme. That struck me as "starts with serious problems like a load of bad bionics or severe drug addiction and no fix" territory, a la the Failed Cyborg and Tweaker. Even the Shower Victim is only -1. (That, and the jumpsuit thing, once Jakers gets back to me about it.)

WRT the Teacher's point cost, they also lack a knife, fire, or storage item (their clothes look like they add up to volume 7). Would factoring that in get them back down to 0, or not so much?

The Gym Teacher also lacks a knife/fire, but that is otherwise a reasonably solid start with the runner pack and baseball bat, so I'll go ahead and bump them up to 3.

Indeed girls can be JDs (and I wouldn't have it any other way!) but moving the hairpins to the gendered stuff would grant a bit of an unfair advantage to female JDs. On the other hand, it occurred to me that the game doesn't specify you're wearing them in your hair: I'm now picturing the JD clipping them to the hem of their socks or a pocket or other such thing inside their jacket. Even if the hairpins appeared in the general inventory rather than as an equipped item, it occurs to me most players would probably wear them right away anyway for convenience; I know I do. Between the "doesn't specify they're being worn in the hair" and "players will wear them in any case" things, I think I'll actually leave this one as it is! Thanks for hammering it out with me, though. :)

Will drop the Student of Faith in the next commit. Might reintroduce it if it's specifically asked for, but I doubt that'll happen.

Were you saying that dropping the chem textbook isn't a bad idea, or that giving them the chem textbook in the first place wasn't a bad idea? Vo-Tech Student's a nice idea, and I'm going to assume that the regional schools in Cata can take Vo-Tech Students since I don't imagine you would have suggested it otherwise. I'll add those in a future commit. Any recommendations on starting items?

Agree on Unprepared Student sounding iffy. How about Poor Student, though? Will that work, or should I come up with a better way to phrase it?

I always pictured jumpsuits as being stereotypical janitor dress; after a brief check, it appears Google likes its pictures of janitors in jumpsuits. If Jakers approves it, I think I'll do it.

AllisonBW commented:
All right, upped cost on Gym Teacher, removed Student of Faith, and played with Science Club Member a bit (removed chemistry set and safety glasses; added a rank of Mechanics and a backpack, because I'd actually intended for them to have a backpack).

KA101 commented:
Standard Teacher was getting a point break already by only paying 1 point for two points' worth of skill.  Speech is one of those tricky ones because it's situational: turn on NPCs? talk them into giving you stuff or running escort, but w/o them it's a Focus sink.  I'd prefer to charge a point, but the lack of survival gear for an immediate bug-out argues in favor of 0.

Science student would have the textbook (whether they could actively read it for any benefit is another story), sure.  That's not a problem.  Backpack gets Iffy again though.

Vo-tech kids are more junior-high/HS students here, so that doesn't fit now that I think about it.  If we had a high school they'd operate out of that as "home base" and then spend about half their week at the tech school, separate building.  (Disclaimer: KA101 is in Pennsylvania, and New England states may not use that sort of system).

Glad the JD worked out.  :-)

Poor Student kinda works.  I want to call it for confusion (student's grades weren't good, or student didn't have much money?) but "Homeless Student"*  or somesuch is overdoing it.  :-/

AllisonBW commented:
If you think the Teacher actually can be left at 0 due to the lack of survival gear, then I'll probably do that, yeah. Good point on being able to use Speaking to ask your starting NPC for gear if you've got NPCs on, though.

I figured science students would be somewhat comparable to the A/V students in this regard; I removed the safety glasses as well as the chemistry set because I was comparing the two. The science student admittedly has a higher-end book, but they won't get much use out of it in the short term; meanwhile, the A/V student has a magazine, but it has several recipes that are more applicable to the early game. Sadly there's not really an equivalent of Ham Radio Illustrated that I can give them; Popular Mechanics only has a recipe for a firearm repair kit, and Under the Hood actually raises skill all the way to 3. (If you'd find Under the Hood preferable to the chemistry textbook, though, I can do that, or just drop the book entirely if you prefer.)

Too bad about the vo-tech students. The idea intrigued me. :( You did mention the regional schools might go up to junior high, though; would that be enough overlap?

As for the Poor Student, I'd make it fairly clear in the flavour blurb: low-income family, on the free school lunch program, etc. If that helps.

KA101 commented:
Chem book is fine, leave it.  Vo-tech is a HS thing generally, last I knew.

Looking forward to Poor Student.

KA101 commented:
Welcome to the open source community, AllisonW.
